traits/ctraits.c

```diff
@@ -4223,34 +4229,60 @@ _trait_delegate(trait_object *trait, PyObject *args)
 |  Sets the appropriate value comparison mode flags of a CTrait instance:
 +----------------------------------------------------------------------------*/
 
-static PyObject *
-_trait_comparison_mode(trait_object *trait, PyObject *args)
+static int
+_set_trait_comparison_mode(trait_object *trait, PyObject *value, void *closure)
 {
-    int comparison_mode;
-
-    if (!PyArg_ParseTuple(args, "i", &comparison_mode)) {
-        return NULL;
+    long comparison_mode = PyLong_AsLong(value);
+    
+    if (comparison_mode == -1 && PyErr_Occurred() != NULL) {
+        return -1;
     }
 
-    trait->flags &= ~(TRAIT_NO_VALUE_TEST | TRAIT_OBJECT_ID_TEST);
     switch (comparison_mode) {
         case 0:
+            trait->flags &= ~TRAIT_COMPARE_MASK;
             trait->flags |= TRAIT_NO_VALUE_TEST;
             break;
         case 1:
+            trait->flags &= ~TRAIT_COMPARE_MASK;
             trait->flags |= TRAIT_OBJECT_ID_TEST;
             break;
         case 2:
+            trait->flags &= ~TRAIT_COMPARE_MASK;
```

Comment:
This may seem silly, but it's worth adding the `|= TRAIT_EQUALITY_COMPARE` here, just for readability. Otherwise a reader is left wondering why this branch is different from the others, and has to go elsewhere in the file to figure out that `TRAIT_EQUALITY_COMPARE` is zero. The compiler would almost certainly optimize that line away, so there's no performance concern. (Also, it's possible that the flag constants could change in the future in such a way that `TRAIT_EQUALITY_COMPARE` is nonzero; this code would then be wrong.)

Reply:
I agree and fixed in b5e9e32.